.github/workflows/single-commit-enforcement.yml

```
@@ -0,0 +1,278 @@
+    runs-on: ubuntu-latest
+    name: 🔒 Single Commit Policy Validation
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Validate Single Commit Policy
+        id: validate
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2129:style:42:1: Consider using { cmd1; cmd2; } >> file instead of individual redirects

(shellcheck)